### PR TITLE
*: add a go.mod file and migrate to v3 package

### PR DESCRIPTION
**Pull request by ericchiang**
Adds a go.mod file and moves the package to the "oidc" directory.
Everything in this PR is just a move except:

* Changes to the test script for go.mods.
* Moves oidc/jose.go comments to above the const block to satisfy golint.
* EDIT: TravisCI was taking a while so I also moved to GitHub Actions. No idea how
to trigger it without merging.

Updates #232

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: test
+on: 
+  pull_request:
+    branches:
+      - v3
```
Comment (mfridman) [inline .github/workflows/test.yaml]: What about enabling actions for any PRs and the default branch: `v3` ?

```suggestion
on:
  push:
    branches:
      - v3
  pull_request:
```

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,13 @@
+module github.com/coreos/go-oidc/v3
+
+go 1.14
+
+require (
+	github.com/coreos/go-oidc v2.2.1+incompatible
```
Comment (mfridman) [inline go.mod]: One suggestion is to run:

```
go mod tidy && go mod verify
```

Which will remove v2 references in go.mod and go.sum

File: .github/workflows/test.yaml (hunk 1 above)
Comment (ericchiang) [reply .github/workflows/test.yaml]: sgtm, added

File: go.mod (hunk 2 above)
Comment (ericchiang) [reply go.mod]: woops, yep done

**Pull request merged**